wire-runtime/src/commonMain/kotlin/com/squareup/wire/WireEnclosingType.kt

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (C) 2024 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.wire
+
+/**
+ * When the enclosing message is pruned, Wire still generates a class for that enclosing message to
+ * preserve the class names and structure.
+ *
+ * This marks such messages and is intended for use by tools like R8 to preserve them.
+ *
+ * https://developer.android.com/build/shrink-code
+ */
+annotation class WireEnclosingType
```

Comment:
Late feedback, but does this need a target or a retention specified?

Reply:
Lets do the same retention that we use for `WireField`